tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/Keramelizer.scala

```diff
@@ -10,8 +12,60 @@ import at.forsyte.apalache.tla.lir.transformations.{TlaExTransformation, Transfo
   *
   * @author Igor Konnov
   */
-class Keramelizer(tracker: TransformationTracker) extends TlaExTransformation {
-  override def apply(ex: TlaEx): TlaEx = {
-    ex
+class Keramelizer(gen: UniqueNameGenerator, tracker: TransformationTracker) extends TlaExTransformation {
+  override def apply(ex: TlaEx) = transform(ex)
+
+  /**
+    * Transform an expression recursively, by rewriting some TLA+ expressions in KerA+.
+    *
+    * @return a new expression
+    */
+  def transform: TlaExTransformation = tracker.track {
+    case OperEx(op, args @ _*) =>
+      transformOneLevel(OperEx(op, args map transform :_*))
+
+    case LetInEx(body, defs @ _*) =>
+      def mapDecl(d: TlaOperDecl): TlaOperDecl = {
+        TlaOperDecl(d.name, d.formalParams, transform(d.body))
+      }
+
+      LetInEx(transform(body), defs.map(mapDecl) :_*)
+
+    case ex =>
+      transformOneLevel(ex)
+  }
+
+  /**
+    * Transform an expression without looking into the arguments.
+    * @return a new expression
+    */
+  private def transformOneLevel: TlaExTransformation = {
+    val identity: PartialFunction[TlaEx, TlaEx] = {
```

Comment:
Why not the scala Predef.identity?

Reply:
Good idea. But it did not work, as `Predef.identity` is not a partial function